### PR TITLE
feat: [IDP-339]: Support for env variables in Plugin API

### PR DESCRIPTION
**Pull request by vigneswara-propelo**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/46129)

### CODE REVIEW

**Comment by vigneswara-propelo** [comment]
trigger ut6

**Comment by vigneswara-propelo** [comment]
trigger ut8

**Comment by vigneswara-propelo** [comment]
trigger ut6

**Review by devesh-harness — CHANGES_REQUESTED** [review]
> Reviewed 15 of 15 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46129)* status: 15 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`idp-service/src/main/java/io/harness/idp/migration/IdpBGMigrationDetails.java` line 39 at r1](https://reviewable.io/reviews/harness/harness-core/46129#-NSGxzeb7iOKePMYnwGZ:-NSGxzeb7iOKePMYnwG_:b-drcpk6) ([raw file](https://github.com/harness/harness-core/blob/3bd991c99ddacd7054bcf398b0fd61ab45b17627/idp-service/src/main/java/io/harness/idp/migration/IdpBGMigrationDetails.java#L39)):*
> ```Java
>         .add(Pair.of(3, PluginInfoMigration.class))
>         .add(Pair.of(4, PluginInfoMigration.class))
>         .build();
> ```

is this intentional ??



_Code quote:_
```Java
        .add(Pair.of(4, PluginInfoMigration.class))
        .build();
```
___
*[`idp-service/src/main/java/io/harness/idp/plugin/beans/PluginInfoEntity.java` line 46 at r1](https://reviewable.io/reviews/harness/harness-core/46129#-NSGy5Kf4BO2qyr1bpcQ:-NSGy5Kf4BO2qyr1bpcR:b-q544ei) ([raw file](https://github.com/harness/harness-core/blob/3bd991c99ddacd7054bcf398b0fd61ab45b17627/idp-service/src/main/java/io/harness/idp/plugin/beans/PluginInfoEntity.java#L46)):*
> ```Java
>   private String source;
>   private String config;
>   private List<String> tokens;
> ```

Lets rename it to envVariables for uniformity.

File: idp-service/contracts/openapi/v1/openapi.yaml [hunk 1]
```diff
@@ -1062,10 +1062,24 @@ components:
           $ref: '#/components/schemas/Exports'
           x-stoplight:
             id: ns1ky9089aalt
+        env_variables:
+          x-stoplight:
+            id: 4tj9fofa4b3hn
+          type: array
+          items:
+            $ref: '#/components/schemas/EnvVariableMap'
```
Comment (vikyathharekal) [inline idp-service/contracts/openapi/v1/openapi.yaml:1070]: Can we use BackstageEnvSecretVariable instead of creating a new one?

**Review by vikyathharekal — CHANGES_REQUESTED** [review]

**Review by vigneswara-propelo — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46129)* status: 16 file(s) to review. 2 change requests. 3. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`idp-service/contracts/openapi/v1/openapi.yaml` line 1070 at r1](https://reviewable.io/reviews/harness/harness-core/46129#gh-1159305414:-NSLEu5R9SSIgr4nrhJW:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/3bd991c99ddacd7054bcf398b0fd61ab45b17627/idp-service/contracts/openapi/v1/openapi.yaml#L1070)):*
<details><summary><i>Previously, vikyathharekal (Vikyath Harekal) wrote…</i></summary><blockquote>

Can we use BackstageEnvSecretVariable instead of creating a new one?
</blockquote></details>

Done.
___
*[`idp-service/src/main/java/io/harness/idp/migration/IdpBGMigrationDetails.java` line 39 at r1](https://reviewable.io/reviews/harness/harness-core/46129#-NSGxzeb7iOKePMYnwGZ:-NSK9hqSDqR-FVjDsj4t:bz67fw5) ([raw file](https://github.com/harness/harness-core/blob/3bd991c99ddacd7054bcf398b0fd61ab45b17627/idp-service/src/main/java/io/harness/idp/migration/IdpBGMigrationDetails.java#L39)):*
<details><summary><i>Previously, devesh-harness wrote…</i></summary><blockquote>

is this intentional ??
</blockquote></details>

yes, if there is any change in the plugin metadata we need to increment this number to trigger migration
___
*[`idp-service/src/main/java/io/harness/idp/plugin/beans/PluginInfoEntity.java` line 46 at r1](https://reviewable.io/reviews/harness/harness-core/46129#-NSGy5Kf4BO2qyr1bpcQ:-NSLEwxz6Oz5J-2ECQhj:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/3bd991c99ddacd7054bcf398b0fd61ab45b17627/idp-service/src/main/java/io/harness/idp/plugin/beans/PluginInfoEntity.java#L46)):*
<details><summary><i>Previously, devesh-harness wrote…</i></summary><blockquote>

Lets rename it to envVariables for uniformity.
</blockquote></details>

Done.

**Review by devesh-harness — APPROVED** [review]
:lgtm:

> Reviewed 11 of 11 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46129#-:-NSLfzXm0cZSBax7Y0SG:bnfp4nl)* status: 16 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by mankrit-singh — APPROVED** [review]
> Reviewed 5 of 15 files at r1, 11 of 11 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46129)* status: 1. unresolved discussions.

**Pull request merged**